### ISSUE TITLE
document shuffling vs non shuffling affects document retrieval per topic

### ISSUE THREAD

**Issue by iamdank** [ddangelov/Top2Vec#300]
![2B1D9FB5-A702-48F7-AA24-DC6BE67E641A](https://user-images.githubusercontent.com/68849237/201636521-0ff50e5e-6a54-45d4-a975-07a0b72e03d9.jpeg)

I trained two models: with and without shuffling of the 20news dataset.(same data, different order).

the top cluster words are very similar, however the associated top documents are quite different.

the top 100 documents for cluster 0’s had only one document overlap. top 1000 for cluster 0’s had about 6% overlap. 

have i made an error in my implementation or overlooked something? i’ve been stuck on this for awhile

**Comment by iamdank** [comment]
code from screenshot example:

umap_args = {
    "n_neighbors": 15,
    "n_components": 5, # 5 -> 2 for plotting 
    "metric": "cosine",
    "random_state": 42 
}

from top2vec import Top2Vec
from sklearn.datasets import fetch_20newsgroups

newsgroups = fetch_20newsgroups(subset='all', shuffle=False, random_state=42, remove=('headers', 'footers', 'quotes'))
model0 = Top2Vec(documents=newsgroups.data, embedding_model="all-MiniLM-L6-v2", umap_args=umap_args)

newsgroups = fetch_20newsgroups(subset='all', shuffle=True, random_state=42, remove=('headers', 'footers', 'quotes'))
model42 = Top2Vec(documents=newsgroups.data, embedding_model="all-MiniLM-L6-v2", umap_args=umap_args)


print('model0 (no shuffle) num topics: ' + str(model0.get_num_topics() ))
print('model42 (yes shuffle) num topics: ' + str(model42.get_num_topics() ))
print('model0 (no shuffle) topic_words [0]: \n ' + str(model0.topic_words[0]) )
print('model0 (yes shuffle) topic_words [0]: \n ' + str(model42.topic_words[0]) )


documents, document_scores, document_ids = model0.search_documents_by_topic(topic_num=0, num_docs=100)
noshuf_document_ids_top0 = document_ids
documents, document_scores, document_ids = model42.search_documents_by_topic(topic_num=0, num_docs=100)
shuf_document_ids_top0 = document_ids


print ('noshuf_document_ids_top0 \n' + str(noshuf_document_ids_top0) + '\n')
print ('shuf_document_ids_top0 \n' + str(shuf_document_ids_top0))


import numpy as np
intersect = np.intersect1d(noshuf_document_ids_top0,shuf_document_ids_top0)
intersect

**Comment by iamdank** [comment]
Fixed... I neglected the document_ids.
Imported 20news to a dataframe and defined 'index' (documentID) column as row_nums +1. 

df['text'] is unshuffled. 
df42['text'] is shuffled rows.   
document_ids set to either the df[index].to_numpy or df42[index].to_numpy

for cluster 0's (comparison between shuffled and non-shuffled):
- top 10 topics words (No overlap)
- top 100 topic words (99% overlap)  
- top 1000 topic words (99% overlap)
